app/src/main/java/org/oppia/android/app/home/promotedlist/PromotedStoryListViewModel.kt

```diff
@@ -28,18 +34,48 @@ class PromotedStoryListViewModel(
     else activity.resources.getDimensionPixelSize(R.dimen.home_padding_start)
 
   /** Determines and returns the visibility for the "View All" button. */
+  fun getHeader(): String {
+    return when {
+      recommendedActivityList.recommendedStoryList.suggestedStoryCount != 0 -> {
+        if (recommendedActivityList.recommendedStoryList.recentlyPlayedStoryCount != 0
+          || recommendedActivityList.recommendedStoryList.olderPlayedStoryCount != 0
+        ) {
+          activity.getString(R.string.stories_for_you)
+        } else
+          activity.getString(R.string.recommended_stories)
+      }
+      recommendedActivityList.recommendedStoryList.recentlyPlayedStoryCount != 0 -> {
+        activity.getString(R.string.recently_played_stories)
+      }
+      else -> {
+        activity.getString(R.string.last_played_stories)
+      }
+    }
+  }
+
+  /**
+   * Determines and returns the visibility for the "View All" button.
+   */
   fun getButtonVisibility(): Int {
-    if (activity.resources.getBoolean(R.bool.isTablet)) {
-      when (Resources.getSystem().configuration.orientation) {
-        Configuration.ORIENTATION_PORTRAIT -> {
-          return if (promotedStoryList.size > promotedStoriesTabletSpanCount) View.VISIBLE
-          else View.INVISIBLE
-        }
-        Configuration.ORIENTATION_LANDSCAPE -> {
-          return if (promotedStoryList.size > promotedStoriesTabletSpanCount) View.VISIBLE
-          else View.INVISIBLE
+    when {
+      recommendedActivityList.recommendedStoryList.suggestedStoryCount != 0 -> {
```

Comment:
Can we do something like this 
```
 recommendedActivityList.recommendedStoryList.let{ storyList ->
    storyList.suggestedStoryCount
    storyList.recentlyPlayedStoryCount
}
```
can increase a bit of readability here using `let`

Reply:
It is condition to check whether `suggestedStoryCount` is 0 or more. This is throwing me error. I can write a if stmt to make it simpler. PTAL if it make sense.

Reply:
I wasn't referring to the logical thing here i was only referring to increase the readibility of the code by simplifying the something shorter by using the `let` function does that make sense ?

Reply:
Fixed this thanks.